pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/ParametricRuleViolation.java

```diff
@@ -34,26 +35,39 @@
     // RuleViolationFactory to support identifying without a Node, and update
     // Rule base classes too.
     // TODO we never need a node. We just have to have a "position", ie line/column, or offset, + file, whatever
+
+    /**
+     * @deprecated Use {@link #ParametricRuleViolation(Rule, String, Node, String)}
+     */
+    @Deprecated
```

Comment:
Deprecated or DeprecatedUntil700?
If deprecated, we need to merge this back to master -> API change...